apps/extension/src/routes/popup/home/index-header.tsx

```diff
@@ -1,27 +1,30 @@
 import { HamburgerMenuIcon } from '@radix-ui/react-icons';
 import { usePopupNav } from '../../../utils/navigate';
 import { PopupPath } from '../paths';
-import { NetworksPopover } from '@penumbra-zone/ui/components/ui/networks-popover';
+import { Network } from '@penumbra-zone/ui/components/ui/network';
 import { useChainIdQuery } from '../../../hooks/chain-id';
 import { motion } from 'framer-motion';
+import { useStore } from '../../../state';
 
 export const IndexHeader = () => {
   const navigate = usePopupNav();
   const { chainId } = useChainIdQuery();
+  const frontendUrl = useStore(state => state.connectedSites.frontendUrl);
 
   return (
     <header className='top-0 z-40 w-full'>
-      <div className='flex items-center justify-between pt-4'>
+      <div className='flex items-center justify-between gap-4 pt-4'>
         <HamburgerMenuIcon
           onClick={() => navigate(PopupPath.SETTINGS)}
-          className='size-6 cursor-pointer hover:opacity-50'
+          className='size-6 shrink-0 cursor-pointer hover:opacity-50'
```

Comment:
Accounting for various spacing issues here to make sure that the `<Network />` component fits even when it has the external link icon.